File: web-sources/S8-fwks-palm/workspace/grid/CreateWorkspaceGridCard.js

import { WorkspaceGridCard } from "./WorkspaceGridCard.js";




export class CreateWorkspaceGridCard extends WorkspaceGridCard {

    constructor(){
        super();

        this.containerNode.setAttribute("type", "create");

        this.headerNode = document.createElement("div");
        this.headerNode.classList.add("workspace-grid-card-create-header");
        this.headerNode.innerHTML = "+";
        this.containerNode.appendChild(this.headerNode);

        this.bodyNode = document.createElement("div");
        this.bodyNode.classList.add("workspace-grid-card-create-body");
        this.bodyNode.innerHTML = "Create a new project...";
        this.containerNode.appendChild(this.bodyNode);
    }


    /**
     * 
     * @param {string} cards
     */
    S8_set_isSelected(isSelected) {
        if (isSelected) {
            this.containerNode.setAttribute("selected", '');
        }
        else {
            this.containerNode.removeAttribute("selected");
        }
    }

    S8_render() { /* continuous rendering approach... */ }

    S8_dispose() { /* continuous rendering approach... */ }
    
}

File: sources/com/s8/pkgs/palm/components/workspace/grid/CreateWorkspaceGridCard.java
package com.s8.pkgs.palm.components.workspace.grid;

import com.s8.api.web.S8WebFront;
import com.s8.pkgs.palm.components.PalmWebSources;


/**
 * 
 * 
 * 
 * @author pierreconvert
 *
 */
public class CreateWorkspaceGridCard extends WorkspaceGridCard {
	
	public static CreateWorkspaceGridCard create(S8WebFront branch) {
		CreateWorkspaceGridCard card = new CreateWorkspaceGridCard(branch);
		return card;
	}


	
	/**
	 * 
	 */
	public final static String WEBPATH = PalmWebSources.ROOT_WEBPATH + "/workspace/grid/CreateWorkspaceGridCard";
	
	
	
	/**
	 * 
	 * @param branch
	 */
	public CreateWorkspaceGridCard(S8WebFront branch) {
		super(branch, WEBPATH);
	}
	
	
	
}
